File: tmdb-core/src/main/java/dev/reuss/tmdb/query/TmdbQuery.java
package dev.reuss.tmdb.query;

import dev.reuss.tmdb.core.http.QueryParams;

import java.io.Serializable;

/**
 * Common contract for TMDB query parameter objects.
 */
public interface TmdbQuery extends Serializable {

    QueryParams toQueryParams();
}

File: tmdb-core/src/main/kotlin/dev/reuss/tmdb/query/TmdbQuery.kt
package dev.reuss.tmdb.query

import dev.reuss.tmdb.core.http.QueryParams
import java.io.Serializable

/**
 * Serializable request options that can be converted to TMDB wire parameters.
 */
interface TmdbQuery : Serializable {
    /** Creates a parameter snapshot suitable for a [dev.reuss.tmdb.core.http.TmdbRequest]. */
    fun toQueryParams(): QueryParams
}
